### PR TITLE
Docs(GraphQL) add list coercion documentation and example.

### PR DESCRIPTION
**Pull request by verneleem**
Also fixed a typo from `eg` to `eq` that caught my eye.

This change was implemented in 20.11 to coerce types into lists and specifically allow the `and` and `or` filter to be a list of filters. It was not clearly stated in the changelog, but can be found in these discuss posts:

https://discuss.dgraph.io/t/there-can-be-only-one-input-field-named/8384/9
https://discuss.dgraph.io/t/limitations-of-graphql-nested-filters/10798

### CODE REVIEW

**Comment by verneleem** [comment]
@aaroncarey I added the note and then decided instead to put it in as a new line with it's own example and the end of the existing documentation. The wording should indicate correctly that the aforementioned documentation is still backwards compatible.

File: content/graphql/queries/and-or-not.md [hunk 1]
```diff
@@ -49,6 +49,19 @@ Example: Posts that have "GraphQL" in the title, or have the tag "GraphQL" and m
...
 queryPost(filter: {
     title: { allofterms: "GraphQL"},
-    or: { title: { allofterms: "Dgraph" }, tags: { eg: "GraphQL" } }
+    or: { title: { allofterms: "Dgraph" }, tags: { eq: "GraphQL" } }
+  } ) { ... }
+```
+
+The `and` and `or` filter accept a list of filters. Any non list filter will be coeorced into a list. This provides backwards compatibility while allowing for more complex filters.
+
+Example: Posts that have "GraphQL" in the title and not the tag "GraphQL", or have "Dgraph" in the title and not the tag "Dgraph"
```
Comment (bucanero) [inline content/graphql/queries/and-or-not.md:58]:
```suggestion
Example: Posts that have `GraphQL` in the title and not the tag `GraphQL`, or have `Dgraph` in the title and not the tag `Dgraph`
```

File: content/graphql/queries/and-or-not.md [hunk 2]
```diff
@@ -49,6 +49,19 @@ Example: Posts that have "GraphQL" in the title, or have the tag "GraphQL" and m
 ```graphql
 queryPost(filter: {
     title: { allofterms: "GraphQL"},
-    or: { title: { allofterms: "Dgraph" }, tags: { eg: "GraphQL" } }
+    or: { title: { allofterms: "Dgraph" }, tags: { eq: "GraphQL" } }
+  } ) { ... }
+```
+
+The `and` and `or` filter accept a list of filters. Any non list filter will be coeorced into a list. This provides backwards compatibility while allowing for more complex filters.
```
Comment (aaroncarey) [inline content/graphql/queries/and-or-not.md:56]: filter accept -> filters both accept

Comment (aaroncarey) [inline content/graphql/queries/and-or-not.md:56]: I'd suggest replacing:

 "Any non list filter will be coeorced into a list."
 
  with
  
  "Per the GraphQL specification, non-list filters are coerced into a list." (Clarifies that this is part of GraphQL spec compliance, adds a dash to make "non-list" a semantic "word", and avoids the "will be" construction in favor of present tense, fixes typo in "coerced" present above).

File: content/graphql/queries/and-or-not.md [hunk 3]
```diff
@@ -49,6 +49,19 @@ Example: Posts that have "GraphQL" in the title, or have the tag "GraphQL" and m
...
+
+Example: Posts that have `GraphQL` in the title and not the tag `GraphQL`, or have `Dgraph` in the title and not the tag `Dgraph`
```
Comment (aaroncarey) [inline content/graphql/queries/and-or-not.md:58]: I suggest changing:

"Example: Posts that have `GraphQL` in the title and not the tag `GraphQL`, or have `Dgraph` in the title and not the tag `Dgraph`"

To:

"Example: Query for posts that have `GraphQL` in the title but that lack the `GraphQL` tag, or that have `Dgraph` in the title but lack the `Dgraph` tag."

**Review by aaroncarey — APPROVED** [review]
Approved with suggested edits. Thank you for your review! :)

File: content/graphql/queries/and-or-not.md (hunk 2 above)
Comment (verneleem) [reply content/graphql/queries/and-or-not.md]: done

Comment (verneleem) [reply content/graphql/queries/and-or-not.md]: done

File: content/graphql/queries/and-or-not.md (hunk 3 above)
Comment (verneleem) [reply content/graphql/queries/and-or-not.md]: done

File: content/graphql/queries/and-or-not.md (hunk 2 above)
Comment (verneleem) [reply content/graphql/queries/and-or-not.md]: done

**Pull request merged**